pkgs/stdenv/generic/check-meta.nix

```diff
@@ -440,16 +440,21 @@ let
 
   assertValidity = { meta, attrs }: let
       validity = checkValidity attrs;
-    in validity // {
+      inherit (validity) valid;
+  in validity // {
       # Throw an error if trying to evaluate a non-valid derivation
       # or, alternatively, just output a warning message.
       handled =
-        {
-          no = handleEvalIssue { inherit meta attrs; } { inherit (validity) reason errormsg; };
-          warn = handleEvalWarning { inherit meta attrs; } { inherit (validity) reason errormsg; };
-          yes = true;
-        }.${validity.valid};
-
+        (
+          if valid == "no" then (
+            handleEvalIssue { inherit meta attrs; } { inherit (validity) reason errormsg; }
+          )
+          else if valid == "warn" then (
+            handleEvalWarning { inherit meta attrs; } { inherit (validity) reason errormsg; }
+          )
+          else if valid == "yes" then true
+          else "Unknown validitiy: '${valid}'"
```

Comment:
I think the code should diverge here to preserve the previous semantics

Reply:
What does this mean? I think this error message is better than what we had before.

Reply:
Previously the code would fail, and so now it should too. Yes the error message is better.

From a types perspective, this code either returns `true` or throws an error. Returning a string here instead would be a type error.

Reply:
Ah. It was my intention to keep a `throw` here but that was neglected.